.github/workflows/close_pr.yml

```diff
@@ -0,0 +1,55 @@
+name: Label closed PR as merged and leave a comment
+on:
+  push
+
+jobs:
+  comment-and-label:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            if(!context.payload.commits || !context.payload.commits.length) return;
+            const sha = context.payload.commits[0].id;
+            const commit = (await github.rest.git.getCommit({
+              commit_sha: sha,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            })).data;
+
+            // Looking at the commit message, checks which PR number, if any, was closed by this commit
+            const getClosedPrIfExists = (commit) => {
+              if(!commit || !commit.message) return;
+              const prClosingRegex = /Closes https:\/\/github.com\/facebook\/react-native\/pull\/([0-9]+)|Pull Request resolved: https:\/\/github.com\/facebook\/react-native\/pull\/([0-9]+)/;
+              const prClosingMatch = commit.message.match(prClosingRegex);
+              if(!prClosingMatch || (!prClosingMatch[1] && ! prClosingMatch[2])) return;
+              return prClosingMatch[1] ?? prClosingMatch[2];
+            };
+
+            const closedPrNumber = getClosedPrIfExists(commit);
+            if(!closedPrNumber) return;
+
+            const pr = (await github.rest.pulls.get({
+              pull_number: closedPrNumber,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            })).data;
+
+            // If the PR has already been processed (labeled as Merged), skip it
+            const mergedLabel = "Merged";
+            if(pr.labels && pr.labels.some(label => label.name === mergedLabel)) return;
+
+            github.rest.issues.createComment({
+              issue_number: closedPrNumber,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `This pull request was successfully merged by ${commit.author.name} in **${sha}**.\n\n<sup>[When will my fix make it into a release?](https://github.com/facebook/react-native/wiki/Release-FAQ#when-will-my-fix-make-it-into-a-release) | [Upcoming Releases](https://github.com/reactwg/react-native-releases/discussions/categories/releases)</sup>`
```

Comment:
```suggestion
              body: `This pull request was successfully merged by @${commit.author.name} in **${sha}**.\n\n<sup>[When will my fix make it into a release?](https://github.com/facebook/react-native/wiki/Release-FAQ#when-will-my-fix-make-it-into-a-release) | [Upcoming Releases](https://github.com/reactwg/react-native-releases/discussions/categories/releases)</sup>`
```

Reply:
Thank you for calling this out! The API I was using was actually only returning the full name and email of the commit's author, which is why I left out the @. I scratched my head for a bit trying to figure out why the API response did not match the schema in the docs - turns out, there are two extremely similar APIs that produce pretty different responses: 
<img width="448" alt="image" src="https://user-images.githubusercontent.com/991791/204717717-4daf4617-ddc2-4ff0-8e08-758fbca4dae1.png">
and 
<img width="537" alt="image" src="https://user-images.githubusercontent.com/991791/204717777-0fd9565d-03e8-4930-bec4-ff8964b0c015.png">

Anyway, I've made the update to use the more complete API. I noticed that in some cases, the login is still null, so I will fall back to the full author name if that happens.

Reply:
Oh TIL! Thansk for digging into this.